java-checks/src/main/java/org/sonar/java/checks/PreferStreamAnyMatchCheck.java

```diff
@@ -60,20 +68,60 @@
 
   @Override
   protected List<MethodMatcher> getMethodInvocationMatchers() {
-    return Stream.of("java.util.Optional", "java.util.OptionalInt", "java.util.OptionalLong", "java.util.OptionalDouble")
+    List<MethodMatcher> matchers = new ArrayList<>();
+    Stream.of("java.util.Optional", "java.util.OptionalInt", "java.util.OptionalLong", "java.util.OptionalDouble")
       .map(type -> MethodMatcher.create().typeDefinition(type).name("isPresent").withoutParameter())
-      .collect(Collectors.toList());
+      .forEach(matchers::add);
+    STREAM_TYPES.stream()
+      .map(type -> MethodMatcher.create().typeDefinition(type).name("anyMatch").addParameter("java.util.function.Predicate"))
+      .forEach(matchers::add);
+    return matchers;
   }
 
   @Override
-  protected void onMethodInvocationFound(MethodInvocationTree isPresentMIT) {
+  protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    if (mit.symbol().name().equals("isPresent")) {
```

Comment:
I would extract `mit.symbol().name()` in a dedicated variable to not recompute it twice in worst case scenario.